Add getStartMoment to handle month site-time parsing

sirreal opened the pull request:
@enejb reported that monthly navigation was not working correctly in #17285. This was related to the way dates were being parsed.

This PR adds a method `getStartDate` to the activity log which improves parsing of these site-time dates.

It also adds date string validation to the controller to ensure that invalid dates are not passed to the component.

## Testing
1. Visit https://calypso.live/stats/activity?branch=fix/17285-bad-date-parsing
1. Navigate months
1. Ensure the query is correct for _site-time_ month start and end
   * See queries by executing `localStorage.setItem( 'debug', 'calypso:data-layer:activity' )` in the console.
   * Debug dates timestamps by running `new Date( 1501570799999 ).toISOString()` to see a human readable date.

Make sure you test conditions on a site with an offset (`UTC-8`) and with a timezone (`Los Angeles`) in `/wp-admin/options-general.php`. Also test with the date string present (`startDate=2017-09-01`) and no string.

Fixes #17285

tyxla commented on client/my-sites/stats/activity-log/index.jsx:
```diff
@@ -95,6 +95,34 @@ class ActivityLog extends Component {
+		if ( timezone ) {
+			return moment.tz( startDate, timezone );
+		}
+		if ( null !== gmtOffset ) {
```
Haven't really looked into how this can be done, but do you think we can simplify the `gmtOffset` handling code, probably by using `moment.utcOffset` or something?

sirreal replied:
Trying to parse string dates with offsets is turning out to be really complicated.

There are known issues working with offsets and I've gotten a PR ready to work on that (#17320). It's quite complicated and I'd prefer avoid tackling that in this PR.

tyxla commented on client/my-sites/stats/activity-log/index.jsx:
```diff
@@ -95,6 +95,34 @@ class ActivityLog extends Component {
 		window.scrollTo( 0, 0 );
 	}
 
+	getStartMoment() {
```
Have you thought of separating this one in a separate utility like `adjustMoment`? Some tests could help making sure it works as expected.

sirreal replied:
I agree, `adjustMoment` needs tests (see #17320 and the known issue with gmt offsets).

The hard part is parsing with an offset, and I think once I get that right in #17320, this method should become relatively simple. It's also _very_ tied to this component and its props, it doesn't feel quite right to extract it.

enejb commented:
I have tested this and it works as expected for me!

sirreal commented:
With some help from @seear, we seem to have found a way of parsing dates using moment with an offset which is neither the browser's nor utc.

After working with @seear, I also corrected parsing for the current month (no `startDate` supplied) and added validation to the controller.
